staff/alexadark/task-api/src/data/taskData.js

```diff
@@ -0,0 +1,39 @@
+const tasks = []
+const taskData = {
+
+    list: () => tasks,
+
+    create: (id, text) => tasks.push({id, text}),
+
+    retrieve: (id) => {
+        const task = tasks.find(task => task.id === id)
+        if (task) return task
+        throw Error('Task does not exists')
+    },
+
+    delete: (id)=> {
+        const index = tasks.findIndex(task =>task.id === id)
+        if(index < 0) throw Error ('Task does not exists')
+        tasks.splice(index,1)
+    },
+    update(id,text){
+        const task = this.retrieve(id)
+        task.text = text
+    },
+    deleteAll: ()=>{
+        tasks.length = 0
+    },
+    markDone(id){
```

Comment:
this is not a CONCERN of taskData, but of taskLogic. taskData just manages the storing of tasks, not the logic and neither state changes on data. MOVE this logic to taskLogic. anyway, this method (markDone) is not in the requirements of the challenge.